File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/DescriptorAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface DescriptorAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // Descriptor Params
    default DescriptorAttributes setDynamicMerchantName(String dynamicMerchantName) {
        requestBuilder.addElement("merchant_name", dynamicMerchantName);
        return this;
    }

    default DescriptorAttributes setDynamicMerchantCity(String dynamicMerchantCity) {
        requestBuilder.addElement("merchant_city", dynamicMerchantCity);
        return this;
    }

    default RequestBuilder buildDescriptorParams() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/DescriptorAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface DescriptorAttributes {

    // Descriptor Params
    fun setDynamicMerchantName(dynamicMerchantName: String): DescriptorAttributes {
        requestBuilder.addElement("merchant_name", dynamicMerchantName)
        return this
    }

    fun setDynamicMerchantCity(dynamicMerchantCity: String): DescriptorAttributes {
        requestBuilder.addElement("merchant_city", dynamicMerchantCity)
        return this
    }

    fun buildDescriptorParams(): RequestBuilder {
        return requestBuilder
    }

    companion object {

        val requestBuilder = RequestBuilder("")
    }
}